plugins/restful/RestfulDataProviderDbQuery.php

```diff
@@ -99,7 +99,7 @@ public function __construct(array $plugin, \RestfulAuthenticationManager $auth_m
 
     $this->tableName = $options['table_name'];
     $this->idColumn = $options['id_column'];
-    $this->primary = empty($plugin['primary']) ? NULL : $this->primary = $plugin['primary'];
+    $this->primary = empty($plugin['data_provider_options']['primary']) ? NULL : $this->primary = $plugin['data_provider_options']['primary'];
```

Comment:
yes - once I've added the ``id`` to the public field, any update would fail without this fix.